ggshield/filter.py

```diff
@@ -118,19 +125,57 @@ def leak_dictionary_by_ignore_sha(
     return sha_dict
 
 
-def path_filter_set(top_dir: Path, paths_ignore: Iterable[str]) -> Set[str]:
+def translate_user_pattern(pattern: str) -> str:
+    """
+    Translate the user pattern into a regex. This function assumes that the given
+    pattern is valid and has been normalized beforehand.
+    """
+
+    # Escape each special character
+    pattern = "".join(
+        f"\\{char}" if char in REGEX_SPECIAL_CHARS else char for char in pattern
+    )
+
+    # Handle start/end of pattern
+    if pattern[-1] != "/":
+        pattern += "$"
+    if pattern[0] == "/":
+        pattern = "^" + pattern[1:]
+    else:
+        pattern = "(^|/)" + pattern
+
+    # Replace * and ** sequences
+    pattern = re.sub(r"\\\*\\\*/", "([^/]+/)*", pattern)
+    pattern = re.sub(r"\\\*", "([^/]+)", pattern)
+
+    return pattern
+
+
+def is_pattern_valid(pattern: str) -> bool:
+    return bool(pattern) and not INVALID_PATTERNS_REGEX.search(pattern)
+
+
+def init_excluded_patterns(paths_ignore: Iterable[str]) -> Set[str]:
     """
     filter_set creates a set of paths of the ignored
     entries from 3 sources:
     .gitguardian.yaml
     files in .git
     files ignore in .gitignore
     """
-    filters = set()
-    for ignored in paths_ignore:
-        filters.update({str(target) for target in top_dir.glob(ignored)})
-
-    return filters
+    res = set()
+    for path in paths_ignore:
+        if not is_pattern_valid(path):
+            raise click.ClickException(f"{path} is not a valid exclude pattern.")
+        res.add(translate_user_pattern(path))
+    return res
+
+
+def is_filepath_excluded(filepath: str, excluded_patterns: Set[str]) -> bool:
+    for regex in excluded_patterns:
+        if re.search(regex, filepath):
+            return True
+    return False
```

Comment:
nit: can be simplified to:
```suggestion
    return any(re.search(r, filepath) for r in excluded_patterns)
```

Reply:
I don't like the `any` syntax, I feel it is not more readable. But it is clearly more _pythonic_, let's do that ;)